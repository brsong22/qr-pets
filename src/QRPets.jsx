import React, { Component } from 'react';
import TextLabel from './components/InfoComponent/TextLabel';
import TabbedPage from './components/InfoComponent/TabbedPage';
import Image from './components/ImageComponent/Image';
import './App.css';

const infoString = 'this is test info.\nlorem ipsum something something somethingsomething\n teset test test\nsay something about the animals!!!!!';
const picString = 'pic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\npic pic pic\n';
const vidString = 'youtube.com\nyoutube.com\nyoutube.com\nyoutube.com';

class QRPets extends Component {
  constructor(props) {
    super(props);
    this.state = {
      // foundation: '',
      petName: 'moomoo',
      petId: 1,
    };
  }

  render() {
    const { petName, petId } = this.state;
    return (
      <div className="App">
        <div className="container">
          <h1 className="titleHeader">Adopt a pet today!</h1>
          <div className="section" id="basicInfo">
            <div className="section" id="profilePicture">
              <Image imageClassName="thumbnail" src={`https://s3.us-east-2.amazonaws.com/qr-pets-images/${petName}_${petId}.JPG`} alt={petName} width={100} height={100} />
            </div>
            <div className="section" id="basicTextInfo">
              <TextLabel label="Name" value="MooMoo" />
              <TextLabel label="Age" value="3" />
              <TextLabel label="Breed" value="Guinea Pig" />
            </div>
          </div>
          <br />
          <TabbedPage tabs={['info', 'photos', 'videos']} info={[infoString, picString, vidString]} />
        </div>
      </div>
    );
  }
}

// QRPets.defaultProps = {
//   foundation: 'today',
// };

export default QRPets;
