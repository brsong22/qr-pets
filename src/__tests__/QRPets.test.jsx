
import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import toJson from 'enzyme-to-json';
import QRPets from '../QRPets';

jest.mock('axios');

describe('QRPets', () => {
  let wrapper;
  const match = {
    params: {
      qrId: '10',
    },
  };

  axios.get.mockResolvedValue({
    data: {
      s3Url: 'someUrl',
      info: {
        age: '3',
        breed: 'Guinea Pig',
        description: 'testInfo',
        name: 'MooMoo',
      },
    },
  });

  beforeEach(async () => {
    wrapper = shallow(<QRPets match={match} />);
    await wrapper.instance().componentDidMount();
  });

  it('renders as expected', async () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('fetches profile data based on qrId', async () => {
    expect(axios.get).toHaveBeenCalledWith('/profile/10');
    expect(wrapper.state()).toEqual({
      data: {
        s3Url: 'someUrl',
        info: {
          age: '3',
          breed: 'Guinea Pig',
          description: 'testInfo',
          name: 'MooMoo',
        },
      },
    });
  });
});
