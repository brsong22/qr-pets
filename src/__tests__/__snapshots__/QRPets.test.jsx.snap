// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRPets renders as expected 1`] = `
<div
  className="container"
>
  <h1
    className="titleHeader"
  >
    Adopt a pet today!
  </h1>
  <div
    className="section"
    id="basicInfo"
  >
    <div
      className="section"
      id="profilePicture"
    >
      <Image
        alt="MooMoo"
        h="100"
        height={100}
        imageClassName="thumbnail"
        src="https://qr-pets-images.s3.us-east-2.amazonaws.com/undefined"
        w="100"
        width={100}
      />
    </div>
    <div
      className="section"
      id="basicTextInfo"
    >
      <TextLabel
        label="Name"
        value="MooMoo"
      />
      <TextLabel
        label="Age"
        value="3"
      />
      <TextLabel
        label="Breed"
        value="Guinea Pig"
      />
    </div>
  </div>
  <br />
  <TabbedPage
    info={
      Array [
        "testInfo",
        "pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
pic pic pic
",
        "youtube.com
youtube.com
youtube.com
youtube.com",
      ]
    }
    tabs={
      Array [
        "info",
        "photos",
        "videos",
      ]
    }
  />
</div>
`;
